echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/PipelineCache.java

```diff
@@ -114,4 +117,28 @@ private void onFront50Error(Throwable e) {
     log.error("Error fetching pipelines from Front50: {}", e.getMessage());
     registry.counter("front50.errors").increment();
   }
+
+  // visible for testing
+  public static List<Pipeline> decorateTriggers(final List<Pipeline> pipelines) {
+    return pipelines
+      .stream()
+      .map(p -> {
+        List<Trigger> triggers = p.getTriggers();
+        if (triggers == null) {
+          return p;
+        }
+
+        List<Trigger> newTriggers = new ArrayList<>(triggers.size());
+        Pipeline newPipe = p.withTriggers(newTriggers);
```

Comment:
I've never been a fan of truncating variable names ... `newPipeline` reads better than `newPipe` to me.

could even just do ...

```
.map(pipeline -> {
...

pipeline = pipeline.withTriggers(
  triggers.stream().map { ... }.collect(Collectors.toList())
)

return pipeline
}
```

Reply:
There's a snag though: the new triggers have a reference to the new pipeline, but we don't have a reference to the new pipeline until we make a copy of it with the new list of triggers. So I end up creating the new pipeline with a placeholder triggers object, and then backfilling the new triggers with the reference to the new pipeline.